src/tracer/MOM_generic_tracer.F90

```diff
@@ -423,6 +432,42 @@ subroutine initialize_MOM_generic_tracer(restart, day, G, GV, US, h, tv, param_f
     enddo
     !! end section to re-initialize generic tracers
 
+    call get_param(param_file, "MOM", "PHA_MLD_CALC", CS%mld_pha_calc, &
+                 "If false, use a fixed value for the photoacclimation mixed layer depth within the "//&
+                 "generic tracer update. This MLD is only used for photoacclimation. This variable should "//&
+                 "be set to true if using COBALTv3 for the BGC.", default=.false.)
+    if (CS%mld_pha_calc) then
+      call get_param(param_file, "MOM", "PHA_MLD_USE_DELTA_RHO", CS%mld_pha_use_delta_rho, &
+                   "If true, use a density difference to find the photoacclimation mixed layer depth "//&
+                   "within the generic tracer update. This MLD is only used for photoacclimation.", default=.false.)
+      call get_param(param_file, "MOM", "PHA_MLD_USE_DELTA_ENG", CS%mld_pha_use_delta_eng, &
+                   "If true, use an energy difference to find the photoacclimation mixed layer depth "//&
+                   "with the generic tracer update. This MLD is only used for photoacclimation.", default=.false.)
+      if (CS%mld_pha_use_delta_rho .and. CS%mld_pha_use_delta_eng) then
+        call MOM_error(FATAL, "PHA_MLD_CALC is set to true and PHA_MLD_USE_DELTA_RHO and PHA_MLD_USE_DELTA_ENG "// &
```

Comment:
These error messages force users to specify an option for calculating the PHA_MLD